### ISSUE TITLE
*bug* cid :: Not clearing cached Symbol, when renamed all interfaces names.

### ISSUE THREAD

**Issue by piyushsh** [CKGrafico/inversify-props#16]
Hi there,

So, the issue is that, initially I added singletons for my application such as :

```js
...
container.addSingleton<ISomeService>(SomeService);
container.addSingleton<ISomeOtherService>(SomeOtherService);
....
```

In tests, I mocked these interfaces and it worked fine.

Now, I renamed these services interfaces such as:
```js
container.addSingleton<SomeServiceInterface>(SomeService);
container.addSingleton<SomeOtherServiceInterface>(SomeOtherService);
```
Now, in tests when I have mocked these interfaces and overriding instances of these interfaces, it is throwing error that `could not unbind serviceIdentifier: undefined`. 
![image](https://user-images.githubusercontent.com/10875373/74175014-eefc3700-4c2c-11ea-9787-d4fb5cc0afe8.png)

When I `console.log(cid)`, it gave me the previously registered Service sysmbols i.e. `ISomeService`, instead it should have been updated to `SomeServiceInterface`.

Please look into this.

**Comment by CKGrafico** [comment]
Yep I think this is related with the automagic ids, https://github.com/CKGrafico/inversify-props/blob/master/src/lib/helpers.ts#L12 I'm assuming that an interface has to start with I but it has not too.

The temporal way to fix this is that you use your own Ids (as inversify does) instead of the autogenerated Ids.

I have to think how to be less "automagic" suggestions are accepted

**Comment by CKGrafico** [comment]
I will probably register all the dependencies with and without 'I' that means `@inject(cid.IService) service: IService` and `@inject(cid.Service) service: IService` will give the same result.

I'm not really convinced but I don't have a better idea right now

**Comment by piyushsh** [comment]
@CKGrafico thanks for replying. 

> I will probably register all the dependencies with and without 'I' that means @inject(cid.IService) service: IService and @inject(cid.Service) service: IService will give the same result.

This is not very clean solution, and adds unnecessary + additional dependency to the class, which we shouldn't be doing.

May be using `customId` could do the job. I'll check it tomorrow. 😄

**Comment by CKGrafico** [comment]
Yes I definitely have to make this better, I'm thinking into create a 2.0 version starting from scratch and testing everything better but I probably will need some time. 

Tell me if the customId works for you if not I can do a patch to you before I start to rewrite the library

**Comment by piyushsh** [comment]
Using customID didn't work either, so I jumped back to initial implementation of `IInterfaceName`.

**Comment by CKGrafico** [comment]
Sorry for that, I started today the v2, I hope everything will be easier in this version

**Comment by CKGrafico** [comment]
Hello I just published a version 2.0.0-alpha1 if you want to check you will find the code in v2 branch https://github.com/CKGrafico/inversify-props/tree/v2
